clang/lib/DPCT/MigrationAction.cpp

```diff
@@ -14,11 +14,67 @@
 #include "MigrationRuleManager.h"
 #include "MisleadingBidirectional.h"
 
+#ifdef _WIN32
+#include <Windows.h>
+#else // __WIN32
+#include <fstream>
+#endif // __WIN32
+
 extern bool StopOnParseErr;
 
 namespace clang {
 namespace dpct {
 
+namespace {
+constexpr size_t MinAvailableMemorySize = 512 * 1024 * 1024; // 512Mb
+constexpr size_t MinAvailableMemoryPercent = 25;             // 25 percent
+
+/// Return true if available phy memory is larger than \p
+/// MinAvailableMemorySize and available phy memory percents is larger than \p
+/// MinAvailableMemoryPercent. Otherwise return false.
+bool checkMemoryStatus();
+
+#ifdef _WIN32
+bool checkMemoryStatus() {
+  MEMORYSTATUSEX MStatus;
+  MStatus.dwLength = sizeof(MStatus);
+  GlobalMemoryStatusEx(&MStatus);
```

Comment:
Please check the retrun value of 41,  and if it fails return a predefined default behaiour.